### PR TITLE
bfv updates

### PR DESCRIPTION
**Pull request by jtopjian**
For #98

### CODE REVIEW

**Comment by jtopjian** [comment]
@jrperritt Thoughts on the changes to `SourceType`? Maybe some validation like:

```go
if opts.SourceType == SourceDestinationType.Blank {
  // SourceType cannot be Blank
}
```

In some ways it feels awkward, but having `SourceType` as a defined type and `DestinationType` as a string feels a little awkward, too.

Or is there some way to have `Volume` exist as both that is just outside of my Go knowledge?

**Comment by jtopjian** [comment]
@jrperritt This is ready for review.

Please see the comments in #98 for API compatibility and let me know if there's any more research you'd like me to do.

With regard to the code specifically, the only thing I'm unsure about is the shared Type between source and destination.

Re my other comment about validity, it seems there are so many different combinations that can be done here (including letting `SourceType` be "blank", contrary to what I originally thought above), that it's probably best to just let the Compute service sort it out and return an error ?

**Comment by jrperritt** [comment]
Will review tomorrow

File: openstack/compute/v2/extensions/bootfromvolume/requests.go [hunk 1]
```diff
@@ -5,26 +5,29 @@ import (
...
-	Snapshot SourceType = "snapshot"
+	Snapshot SourceDestinationType = "snapshot"
 	// Image SourceType
-	Image SourceType = "image"
+	Image SourceDestinationType = "image"
 	// Blank SourceType
-	Blank SourceType = "blank"
+	Blank SourceDestinationType = "blank"
+	// Local DestinationType
+	Local SourceDestinationType = "local"
```
Comment (jrperritt) [inline openstack/compute/v2/extensions/bootfromvolume/requests.go]: How about we leave `SourceType` as-is, prefix the current constants with 'Source', and define a new `DestinationType`:
```go
type (
  DestinationType string
  SourceType string
)

const (
  SourceVolume SourceType = "volume"
  DestinationVolume DestinationType = "volume"
  ...
)
```

File: openstack/compute/v2/extensions/bootfromvolume/requests.go [hunk 2]
```diff
@@ -34,11 +37,11 @@ type BlockDevice struct {
 	DeleteOnTermination bool `json:"delete_on_termination"`
 	// DestinationType is the type that gets created. Possible values are "volume"
 	// and "local".
-	DestinationType string `json:"destination_type,omitempty"`
+	DestinationType SourceDestinationType `json:"destination_type,omitempty"`
 	// GuestFormat specifies the format of the block device.
 	GuestFormat string `json:"guest_format,omitempty"`
 	// VolumeSize is the size of the volume to create (in gigabytes).
-	VolumeSize int `json:"volume_size"`
+	VolumeSize int `json:"volume_size,omitempty"`
```
Comment (jrperritt) [inline openstack/compute/v2/extensions/bootfromvolume/requests.go]: :+1:

**Comment by jtopjian** [comment]
@jrperritt ping for review

**Comment by jrperritt** [comment]
will review tomorrow

File: openstack/compute/v2/extensions/bootfromvolume/requests.go [hunk 3]
```diff
@@ -5,40 +5,65 @@ import (
...
-	// Image SourceType
-	Image SourceType = "image"
 	// Blank SourceType
 	Blank SourceType = "blank"
+
+	// Volume DestinationType
+	DestinationVolume DestinationType = "volume"
+
+	// Image SourceType
+	Image SourceType = "image"
```
Comment (jrperritt) [inline openstack/compute/v2/extensions/bootfromvolume/requests.go]: These should all have either a 'Destination' or 'Source' prefix.
Also, let's group them by type, no alphabetically.

**Comment by jrperritt** [comment]
Also, please do **not** rebase after a review has started

**Comment by jtopjian** [comment]
@jrperritt Yeah, that's my bad - I need to break that habit.

Please see 599720ce8db8f65581b9a4072ebc1f15972ca8ef -- does that look OK for the rename / reorder?

**Comment by jrperritt** [comment]
This looks good. +2

**Comment by jtopjian** [comment]
Thank you again for your help and patience :)

**Pull request merged**